go-controller/pkg/ovn/ovn.go

```diff
@@ -780,12 +780,29 @@ func (oc *Controller) WatchNetworkPolicy() {
 		AddFunc: func(obj interface{}) {
 			policy := obj.(*kapisnetworking.NetworkPolicy)
 			oc.initRetryPolicy(policy)
+			oc.checkAndSkipRetryPolicy(policy)
+			// If there is a delete entry this is a network policy being added
+			// with the same name as a previous network policy that failed deletion.
+			// Destroy it first before we add the new policy.
+			if retryEntry := oc.getPolicyRetryEntry(policy); retryEntry != nil && retryEntry.oldPolicy != nil {
+				klog.Infof("Detected stale policy during new policy add with the same name: %s/%s",
+					policy.Namespace, policy.Name)
+				if err := oc.deleteNetworkPolicy(retryEntry.oldPolicy, nil); err != nil {
+					oc.unSkipRetryPolicy(policy)
+					klog.Errorf("Failed to delete stale network policy %s, during add: %v",
+						getPolicyNamespacedName(policy), err)
+					return
```

Comment:
@trozet so what happens if we fail here? We won't call AddFunc() again because the API/informer already has it in the cache, we'll call UpdateFunc() instead when the policy changes. But UpdateFunc() doesn't try to delete retryEntry.oldPolicy at all, so would it just get left hanging around?

Reply:
the update won't be called since the new policy doesn't actually change right? I think... here is what would happen: Since we called initRetryPolicy on L783 we have added the new policy to retry queue. then we call unskipretry in L791 where we set ignore=false, that makes this new policy a candidate for add retry right?